apps/src/applab/dataBrowser/AddTableRow.jsx

```diff
@@ -1,10 +1,32 @@
+import FirebaseStorage from '../firebaseStorage';
 import Radium from 'radium';
 import React from 'react';
+import { castValue, displayValue } from './dataUtils';
 import * as dataStyles from './dataStyles';
 
 const AddTableRow = React.createClass({
   propTypes: {
-    columnNames: React.PropTypes.array.isRequired
+    columnNames: React.PropTypes.array.isRequired,
+    tableName: React.PropTypes.string.isRequired
+  },
+
+  getInitialState() {
+    return { newRecord: {} };
```

Comment:
Per your later comment, this would preclude adding other things to `state` in the future, which I don't love.